### PR TITLE
Upgrade to Rails 6.1

### PR DESCRIPTION
**Pull request by Throne3d**

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -51,9 +53,9 @@
   config.force_ssl = true
   config.ssl_options = { hsts: { preload: true } }
 
-  # Use the lowest log level to ensure availability of diagnostic information
-  # when problems arise.
-  config.log_level = :warn
+  # Include generic and useful information about system operation, but avoid logging too much
+  # information to avoid inadvertent exposure of personally identifiable information (PII).
+  config.log_level = :info
```
Comment (teceler) [inline config/environments/production.rb:58]: I observe this changes our log level on prod (unclear on in which direction); do we want that?

File: config/environments/production.rb [hunk 2]
```diff
@@ -79,11 +81,17 @@
   # Send deprecation notices to registered listeners.
   config.active_support.deprecation = :notify
 
+  # Log disallowed deprecations.
+  config.active_support.disallowed_deprecation = :log
+
+  # Tell Active Support which deprecation messages to disallow.
+  config.active_support.disallowed_deprecation_warnings = []
```
Comment (teceler) [inline config/environments/production.rb:88]: This otoh just seems reasonable tbh

File: config/environments/test.rb [hunk 3]
```diff
@@ -43,8 +45,17 @@
...
+
+  # Tell Active Support which deprecation messages to disallow.
+  config.active_support.disallowed_deprecation_warnings = []
+
   # Raises error for missing translations.
-  # config.action_view.raise_on_missing_translations = true
+  # config.i18n.raise_on_missing_translations = true
+
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
```
Comment (teceler) [inline config/environments/test.rb:58]: what does this... do... I realize it isn't enabled but I'm now curious

File: config/application.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
-require_relative 'boot'
+require_relative "boot"
 
-require 'rails/all'
+require "rails/all"
```
Comment (teceler) [inline config/application.rb:3]: I would probably have culled the quotation-mark changes but that's not actually important either way

File: Gemfile [hunk 5]
```diff
@@ -9,8 +9,8 @@ gem 'apipie-rails'
 # - for migrations with `rails generate audited:upgrade`
 # - that the method set_audit_user has not changed, since we duplicate it in
 #   ApplicationRecord for use in callbacks to send audit user ids to background jobs.
-#   (currently https://github.com/collectiveidea/audited/blob/v4.9.0/lib/audited/audit.rb#L175)
-gem 'audited', '~> 4.9.0'
+#   (currently https://github.com/collectiveidea/audited/blob/v5.0.2/lib/audited/audit.rb#L183)
+gem 'audited', '~> 5.0.0'
```
Comment (teceler) [inline Gemfile:13]: Did we check if this has migrations?

File: config/initializers/new_framework_defaults_6_1.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file contains migration options to ease your Rails 6.1 upgrade.
+#
+# Once upgraded flip defaults one by one to migrate to the new default.
+#
+# Read the Guide for Upgrading Ruby on Rails for more info on each option.
+
+# Support for inversing belongs_to -> has_many Active Record associations.
+# Rails.application.config.active_record.has_many_inversing = true
```
Comment (teceler) [inline config/initializers/new_framework_defaults_6_1.rb:10]: !!!

File: config/storage.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# microsoft:
+#   service: AzureStorage
+#   storage_account_name: your_account_name
+#   storage_access_key: <%= Rails.application.credentials.dig(:azure_storage, :storage_access_key) %>
+#   container: your_container_name
+
+# mirror:
+#   service: Mirror
+#   primary: local
+#   mirrors: [ amazon, google, microsoft ]
```
Comment (teceler) [inline config/storage.yml:34]: I've just been nuking this file when rails migrations add it, but we might actually want to look at activestorage again in 6.1

File: db/migrate/20221012213634_add_service_name_to_active_storage_blobs.active_storage.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# This migration comes from active_storage (originally 20190112182829)
```
Comment (teceler) [inline db/migrate/20221012213634_add_service_name_to_active_storage_blobs.active_storage.rb:1]: We should clear out the activestorage migrations tho

File: db/schema.rb [hunk 9]
```diff
@@ -2,15 +2,15 @@
...
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_04_16_173619) do
+ActiveRecord::Schema.define(version: 2022_10_12_213635) do
```
Comment (teceler) [inline db/schema.rb:13]: Unsure what this is reflecting

**Review by teceler — COMMENTED** [review]
Overall mostly want to check we've got prereqs covered first.

File: config/environments/production.rb (hunk 1 above)
Comment (Throne3d) [reply config/environments/production.rb:58]: We do not, lemme fix it

File: config/application.rb (hunk 4 above)
Comment (Throne3d) [reply config/application.rb:3]: Legit, lemme undo them

File: config/initializers/new_framework_defaults_6_1.rb (hunk 6 above)
Comment (Throne3d) [reply config/initializers/new_framework_defaults_6_1.rb:10]: IKR! Hopefully this will be helpful

File: db/schema.rb (hunk 9 above)
Comment (Throne3d) [reply db/schema.rb]: (the Audited change)

File: config/environments/test.rb (hunk 3 above)
Comment (Throne3d) [reply config/environments/test.rb:58]: It adds comments to the HTML in test mode, like this!

```html
<!-- BEGIN app/views/layouts/partials/_header.html.erb -->
  <header></header>
<!-- END app/views/layouts/partials/_header.html.erb -->
```
Just to help understand what's going wrong - I guess it's useful for view specs, not that we really have those =)

File: config/cable.yml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+development:
+  adapter: async
+
+test:
+  adapter: test
+
+production:
```
Comment (Marri) [inline config/cable.yml:7]: Are we in fact using Cable...?

**Review by Marri — COMMENTED** [review]
I think that's all I saw?

File: config/cable.yml (hunk 10 above)
Comment (teceler) [reply config/cable.yml:7]: pretty sure no, it's just autogenerated

File: README.md [hunk 11]
```diff
@@ -4,7 +4,7 @@
 
 ### Dependencies
 
-*   [Ruby](https://www.ruby-lang.org/en/) 2.7.5
+*   [Ruby](https://www.ruby-lang.org/en/) 2.7.7
 *   [Rails](http://rubyonrails.org/) 6.0
```
[the remark this answers is not in the document]
Comment (Throne3d) [reply README.md]: whoops

File: config/cable.yml (hunk 10 above)
Comment (Throne3d) [reply config/cable.yml:7]: yeah, we shouldn't be - just autogenerated

I can try removing it and hope everything works? :D

**Review by Marri — APPROVED** [review]

**Pull request merged**